### PR TITLE
[Kotlin Migration] AbstractFlashcardViewer

### PR DESCRIPTION
**Pull request by david-allison**
This one's big, sorry.

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt [hunk 1]
```diff
@@ -0,0 +1,2573 @@
...
+    @JvmField
+    protected var mTopBarLayout: RelativeLayout? = null
+    private val mClipboard: ClipboardManager? = null
+    private var mPreviousAnswerIndicator: PreviousAnswerIndicator? = null
+
+    /** set via [.setCurrentCard]  */
+    @JvmField
+    var mCurrentCard: Card? = null
+    private var mCardSoundConfig: CardSoundConfig? = // set when mCurrentCard is
+        null
```
Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt:189]:
```suggestion
    // set when mCurrentCard is
    private var mCardSoundConfig: CardSoundConfig? = null
        
```

**Comment by Akshay0701** [comment]
have you tested the app? does any part of the code give nullpointerexception?

**Comment by david-allison** [comment]
> have you tested the app? does any part of the code give nullpointerexception?

I've had a brief test and it seemed fine, but there's a lot of moving parts here

**Comment by david-allison** [comment]
Thanks so much for the review!!

**Review by mikehardy — APPROVED** [review]
wow, just wow. Thanks for taking that on.
I don't have any quibbles, if it's functioning, I saw the KotlinCleanups and agreed with them, I think move forward before entropy takes hold, and iterate with GSoC-powered KotlinCleanup (hopefully)

**Review by Akshay0701 — APPROVED** [review]
Great! LGTM

**Pull request merged**